server/src/main/kotlin/delta/codecharacter/server/user/public_user/PublicUserService.kt

```diff
@@ -71,11 +69,50 @@ class PublicUserService(@Autowired private val publicUserRepository: PublicUserR
                     wins = it.wins,
                     losses = it.losses,
                     ties = it.ties
-                )
+                ),
             )
         }
     }
 
+    fun updateLeaderboard(leaderboardList: List<LeaderboardEntryDto>) {
+        val leaderboardSize = leaderboardList.size
+        for (i in 0 until leaderboardSize) {
+            val user = publicUserRepository.findByUsername(leaderboardList[i].user.username).get()
+            if (i < 0.2 * leaderboardSize) {
```

Comment:
instead of having some % of player we can have some fixed number people like top15,top20 . make a variable in `application.yml` and import from there.